server/src/main/java/org/opensearch/index/translog/transfer/TransferService.java

```diff
@@ -114,4 +107,8 @@ void uploadBlobAsync(
      */
     InputStream downloadBlob(Iterable<String> path, String fileName) throws IOException;
 
+    void listAllInSortedOrder(Iterable<String> path, int limit, ActionListener<List<BlobMetadata>> listener) throws IOException;
```

Comment:
Do we need to pass the `ActionListener` for non-async cases

Reply:
Non-async will need to use `Latches` and make it async at the caller side.  This is consistent with `BlobContainer` interface as well .  If we are to provide a method of non async cases, BlobContainer would be the right place for this IMO .